Update guidelines with more information

johlju opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dsccommunity/dsccommunity.org/80)
<!-- Reviewable:end -->

johlju commented:
@gaelcolas can you look at the text to see if I don't lie - and if there are more that could be explained in the areas I touched now.

gaelcolas commented on content/guidelines/administration.md:
```diff
@@ -6,14 +6,110 @@ author: "Johan Ljunggren"
+the @dscbot account (the DSC Community GitHub account) and from there
+take out a Personal Access Token for the the **specific repository**.
+
+**Do NOT EVER create a PAT for the entire organization!**
```
Organisations can't have PAT. Only OAuth token I think.
We are using the DSC Bot account to have a PAT.
Perms are directly coming from the DSC Bot account, so it's push perms on all repos.
We use the PAT to create the PR & to create the GH releases.

johlju replied:
Discussing in Slack. Wait for the conclusion of that discussion so I know how to change this.

johlju replied:
Done

gaelcolas commented on content/guidelines/administration.md:
```diff
@@ -6,14 +6,110 @@ author: "Johan Ljunggren"
+
+### Create the PowerShell Gallery API key
+
+Not yet written.
```
This is where we create 1 API per Package, that's only allowed to push new versions of packages.

gaelcolas replied:
btw, I have the screenshots needed. So as soon as this PR is merged, I can populate this section.

johlju replied:
Cool! I leave this for you to fill in then.

gaelcolas commented on content/guidelines/maintaining.md:
```diff
@@ -29,14 +29,47 @@ As a maintainer you are allowed to review and merge your own code, but
 
 Not written.
 
+>**NOTE:** There are an issue with module ModuleBuilder when using
```
with ModuleBuilder version > 1.0.0

gaelcolas commented on content/guidelines/testing-guidelines.md:
```diff
@@ -68,7 +68,58 @@ to do anything to get new functionality in the DSC modules.
+to pass then there are different scenarios supported to opt-out from tests.
+Preferably you should opt-out from tests.
+
+- Using the opt-out JSON file `.MetaTestOptOut.json`.
```
did you not say we should just put this in Build.yaml and not even talk about that feature (so we can deprecate and remove when we feel like it?).

johlju replied:
Good point. Changing.

gaelcolas commented on content/guidelines/testing-guidelines.md:
```diff
@@ -68,7 +68,58 @@ to do anything to get new functionality in the DSC modules.
+
+##### Test "New Error-Level Script Analyzer Rules"
+
+This HQRM tests is opt-out by default since it tests the ScriptAnalyzer
```
I should fix that in the config file of DscResource.Test.
Will try to do tomorrow.

johlju replied:
Leaving it in for now and we remove it when it changes.

johlju commented on content/guidelines/administration.md:
```diff
@@ -6,14 +6,110 @@ author: "Johan Ljunggren"
+All repositories should always be added with **MIT licensing**.
+
+>**NOTE:** To be able create repositories you must be an owner of the
+>GitHub DSC Community organization. See [GitHub DSC Community organizational owners](https://github.com/orgs/dsccommunity/people?query=role%3Aowner).
```
Discussing in Slack. Wait for the conclusion of that discussion before changing this.

johlju commented:
@gaelcolas moved and updated the section "Create the GitHub Personal Access Token (PAT)". I added a little text to "Create the PowerShell Gallery API key" but you can fill in that with more text later as you mentioned in a review comment.

johlju commented:
@gaelcolas ready for review again.

johlju commented:
Fix the comment. Merging.
